client/browser/src/browser-extension/scripts/optionsPage.main.tsx

```diff
@@ -140,6 +141,25 @@ function useTelemetryService(sourcegraphUrl: string | undefined): TelemetryServi
     return telemetryService
 }
 
+const fetchUserSettingsURL = (sourcegraphURL: string): Observable<string> => {
+    const requestGraphQL = createRequestGraphQL(sourcegraphURL)
+
+    return requestGraphQL<GQL.IQuery>({
+        request: gql`
+            query UserSettingsURL {
+                currentUser {
+                    settingsURL
+                }
+            }
+        `,
+        variables: {},
+    }).pipe(
+        map(({ data }) => data?.currentUser?.settingsURL),
+        filter(Boolean),
+        map(url => new URL(`${url as string}/repositories/manage`, sourcegraphURL).href)
```

Comment:
```suggestion
        filter(isDefined),
        map(url => new URL(`${url}/repositories/manage`, sourcegraphURL).href)
```

**Suggestion**: We should be able to use the [`isDefined`](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@main/-/blob/client/common/src/types/utils.ts?L4%3A20=&utm_source=VSCode-2.2.0) type guard here so we can remove the type assertion. There's little to no risk of `settingsURL` type changing, but it would be ideal to use type guards to safeguard against that consistently (sad that TS can't infer from `filter(Boolean)` correctly, though)

Reply:
I like this suggestion! Thanks @tjkandala!